src/NServiceBus.AcceptanceTests/Routing/When_publishing_to_scaled_out_subscribers_on_unicast_transports.cs

```diff
@@ -0,0 +1,129 @@
+﻿namespace NServiceBus.AcceptanceTests.Routing
+{
+    using System.Threading;
+    using System.Threading.Tasks;
+    using NServiceBus.AcceptanceTesting;
+    using NServiceBus.AcceptanceTesting.Customization;
+    using NServiceBus.AcceptanceTests.EndpointTemplates;
+    using NServiceBus.AcceptanceTests.ScenarioDescriptors;
+    using NServiceBus.Routing;
+    using NUnit.Framework;
+
+    public class When_publishing_to_scaled_out_subscribers_on_unicast_transports : NServiceBusAcceptanceTest
+    {
+        [Test]
+        public async Task Each_event_should_be_delivered_to_single_instance_of_each_subscriber()
+        {
+            await Scenario.Define<Context>()
+                .WithEndpoint<Publisher>(b => b.When(c => c.SubscribersCounter == 4, async (session, c) =>
```

Comment:
@andreasohlund I have a follow up discussion I want to have with you.
Elevator pitch: I think we should get rid of message driven pubsub entirely

On Monday, February 15, 2016, Andreas Öhlund <notifications@github.com>
wrote:

> In
> src/NServiceBus.AcceptanceTests/Routing/When_publishing_to_scaled_out_subscribers_on_unicast_transports.cs
> <https://github.com/Particular/NServiceBus/pull/3459#discussion_r52897816>
> :
>
> > +    using System.Threading;
> > +    using System.Threading.Tasks;
> > +    using NServiceBus.AcceptanceTesting;
> > +    using NServiceBus.AcceptanceTesting.Customization;
> > +    using NServiceBus.AcceptanceTests.EndpointTemplates;
> > +    using NServiceBus.AcceptanceTests.ScenarioDescriptors;
> > +    using NServiceBus.Routing;
> > +    using NUnit.Framework;
> > +
> > +    public class When_publishing_to_scaled_out_subscribers_on_unicast_transports : NServiceBusAcceptanceTest
> > +    {
> > +        [Test]
> > +        public async Task Each_event_should_be_delivered_to_single_instance_of_each_subscriber()
> > +        {
> > +            await Scenario.Define<Context>()
> > +                .WithEndpoint<Publisher>(b => b.When(c => c.SubscribersCounter == 4, async (session, c) =>
>
> On multicast transports subscribe happens in the FST and is synchronous
>
> Can we really assume this?
>
> eg. exchange.bind in rabbit can be async if you use the no-wait flag?
>
> http://www.rabbitmq.com/amqp-0-9-1-reference.html#exchange.bind.no-wait
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Particular/NServiceBus/pull/3459/files#r52897816>.
>